packages/blocks/src/api/factory.js

```diff
@@ -344,6 +344,43 @@ export function getPossibleBlockTransformations( blocks ) {
 	] );
 }
 
+/**
+ * Returns a subset of the transforms from `getPossibleBlockTransformations`.
+ * This subset was selected to support transforms that the mobile apps support,
+ * and to avoid "confusing" transforms that don't feel intuitive in a mobile
+ * context.
+ *
+ * @param {Array} block The block to transform.
+ *
+ * @return {Array} Block types that the block argument can be transformed to in
+ * the mobile app.
+ */
+export function getMobileBlockTransformations( block ) {
+	// NOTE (blixt): For now, we only support transforming *from* the paragraph
+	// block. We'll add others soon!
+	const eligibleOriginBlocks = [ 'core/paragraph' ];
+
+	if ( ! eligibleOriginBlocks.includes( block.name ) ) {
+		return [];
+	}
+
+	// NOTE (blixt): From what I can gather, we want to support most-any
+	// transform, except that we don't want to support transforming into a
+	// column or group. If the rules are more nuanced, we could have a mapping
+	// of disallowed transformations - but for now, let's do a simple blocklist!
+	const ineligibleDestinationBlocks = [ 'core/column', 'core/group' ];
```

Comment:
> Want to ban specific block types from transforming to core/column.

This would align more with what we discussed on our side conversation that we might want to ban specific options.

Reply:
We also probably only want to support our list of supported blocks which can be found here: https://github.com/WordPress/gutenberg/blob/47feef0a2aaaa38b34fb3525a35cfddaf032c719/packages/block-library/src/index.native.js#L195-L227